src/js/index.js

```diff
@@ -0,0 +1,133 @@
+});
+
+window.addEventListener('DOMContentLoaded', () => {
+  const language = localStorage.getItem('language');
+  if (language !== null) {
+    state.lang = language;
+  }
+});
+
+window.addEventListener('DOMContentLoaded', () => {
+  createHeader();
+  createTextarea();
+  board.createKeyBoard(state.lang);
+  createFooter();
+
+  const keyBoard = document.querySelector('.keyboard-list');
+
+  document.addEventListener('keydown', (event) => {
+    if (event.ctrlKey && event.altKey) {
+      if (state.lang === 'ru') {
+        state.lang = 'en';
+      } else {
+        state.lang = 'ru';
+      }
+      keyBoard.innerHTML = '';
+      keyBoard.append(board.generateButtons(state.lang));
+    }
+  });
+
+  const textarea = document.querySelector('.textarea');
+  const capsLock = document.querySelector('.capslock');
+
+  const capsLockActive = () => {
+    const buttons = document.querySelectorAll('.keyboard__button');
+    state.count += 1;
+    state.capsLock = true;
+    buttons.forEach((button) => {
+      const currentButtons = button;
+      if (currentButtons.textContent.length === 1) {
+        currentButtons.textContent = button.textContent.toUpperCase();
+        currentButtons.dataset.keyCode = button.textContent.toUpperCase();
+        capsLock.classList.add('keyboard__button_active');
+      }
+    });
+  };
+  const capsLockUnActive = () => {
+    const buttons = document.querySelectorAll('.keyboard__button');
+    state.count += 1;
+    state.capsLock = false;
+    buttons.forEach((button) => {
+      const currentButtons = button;
+      if (currentButtons.textContent.length === 1) {
+        currentButtons.textContent = button.textContent.toLowerCase();
+        currentButtons.dataset.keyCode = button.textContent.toLowerCase();
+        capsLock.classList.remove('keyboard__button_active');
+      }
+    });
+  };
+
+  document.addEventListener('click', (event) => {
+    const currentButton = event.target.dataset.keyCode;
+    if (currentButton === 'Backspace') {
+      state.text = state.text.slice(0, state.text.length - 1);
+    }
+    if (currentButton === 'Tab') {
+      state.text += '  ';
+    }
+    if (currentButton === ' ') {
+      state.text += ' ';
+    }
+    if (currentButton === 'Enter') {
+      state.text += '\n';
+    }
+    if (currentButton === 'Delete') {
+      const cursorAt = textarea.selectionStart;
+      state.text = state.text.slice(0, cursorAt) + state.text.slice(cursorAt + 1);
+    }
+    if (currentButton === 'CapsLock') {
+      if (!state.capsLock) {
+        capsLockActive();
+      } else {
+        capsLockUnActive();
+      }
+    } else if (event.target.textContent.length === 1) {
+      const buttonValue = event.target.textContent;
+      state.text += buttonValue;
+    }
+    textarea.value = state.text;
+    textarea.focus();
+  });
+
+  document.addEventListener('keydown', (event) => {
+    const buttons = document.querySelectorAll('.keyboard__button');
+    buttons.forEach((button) => {
```

Comment:
It seems to me like you don't need to execute some code or check for each element in buttons. If that's correct, it'll better to use find method to get the specific button and adjust it on key press.